crates/oxc_linter/src/rules/eslint/no_useless_rename.rs

```diff
@@ -0,0 +1,395 @@
+use oxc_ast::{
+    ast::{
+        AssignmentTarget, AssignmentTargetPattern, AssignmentTargetProperty, BindingPatternKind,
+    },
+    AstKind,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+use oxc_span::GetSpan;
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint(no-useless-rename): Disallow renaming import, export, and destructured assignments to the same name")]
+#[diagnostic(severity(warning), help("Either remove the renaming or rename the variable."))]
+struct NoUselessRenameDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoUselessRename(Box<NoUselessRenameConfig>);
+
+#[allow(clippy::struct_field_names)]
+#[derive(Debug, Default, Clone)]
+pub struct NoUselessRenameConfig {
+    ignore_destructuring: bool,
+    ignore_import: bool,
+    ignore_export: bool,
+}
+
+impl std::ops::Deref for NoUselessRename {
+    type Target = NoUselessRenameConfig;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Disallow renaming import, export, and destructured assignments to the same name.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// It is unnecessary to rename a variable to the same name.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // Bad
+    /// import { foo as foo } from 'foo';
+    /// const { bar: bar } = obj;
+    /// export { baz as baz };
+    ///
+    /// // Good
+    /// import { foo } from 'foo';
+    /// const { bar: renamed } = obj;
+    /// export { baz };
+    /// ```
+    NoUselessRename,
+    style
```

Comment:
This isn't marked as style in eslint, and it feels like a `correctness` rule because this is "useless".

Reply:
yep that makes sense to me.